apisix/admin/plugins.lua

```diff
@@ -60,16 +70,30 @@ function _M.get(name)
         return 200, http_plugins
     end
 
-    if not name then
-        return 400, {error_msg = "not found plugin name"}
+    local plugin
+    -- By default search through all subsystems
+    local all_subsystem = true
+    if subsystem then
+        all_subsystem = false
+    end
+
+    -- When subsystem is passed as http or not passed at all(searching through all)
+    if all_subsystem or subsystem == "http"  then
+        plugin = plugin_get_http(name)
     end
```

Comment:
It's a bit complicated here, can we just make it simple like this
```
    if subsystem == "stream" then 
        plugin = plugin_get_stream(name)
    else
        plugin = plugin_get_http(name)   
    end
```

Reply:
No. The default is to filter through both subsystems unless a specific subsystem is specified via query params. This is written in the proposal as well. To make things explicit.